codelabs/codelab-03/README.md

```diff
@@ -0,0 +1,144 @@
+- You'll set up CloudFront via the AWS GUI
+- You'll test latency 
+- You'll issue an invalidation
+
+### Setting Up
+
+Before starting this codelab, run `git pull` in the `389Lspring18` directory to update your local copy of the class repository.
+
+### Tutorial
+
+**It's a good idea to read/skim the entire codelab first so you have an idea of what you're doing. Keep in mind that you will be submitting screenshots of some command outputs, so make sure to read the submission section (at a minimum) located near the bottom to understand what you will be turning in**
+
+#### Back to buckets
+
+Fist, download the image ![canyon.jpg](../../../media/codelabs/codelab-03/canyon.jpg)
+
+Now, using the [AWS GUI](https://console.aws.amazon.com/console/home?region=us-east-1#), navigate to S3.
+
+From the S3 dashboard, click `Create Bucket`.
+
+In the field labeled *bucket name*, use the format `cmsc389l-<your directory id>-codelab-03`.
+
+In the field labeled *region*, select `Asia Pacific (Sydney)`.
+
+This is what your form should look like.
+
+![create bucket Page](../../../media/codelabs/codelab-03/create-bucket.png)
+
+Click the `Create` button in the lower-left corner. For the purposes of this tutorial, the default properties and permissions are fine. Go ahead and skip past the rest of the configuration screens and create the bucket.
+
+Select your newly-created bucket (you have to click the bucket name itself).
+
+Upload `canyon.jpg` from where you downloaded it on your machine. By default, S3 objects permissions are set to private. We will need to modify those permissions to public.
+
+Click `Next` and select `Grant public read access to this object(s)` from the drop-down menu labeled *Manage public permissions*.
+
+Your form should look similar to this.
+
+![public-read-access Page](../../../media/codelabs/codelab-03/public-read-access.png)
+
+Click the `Upload` button in the lower-left corner.
+
+#### Testing latency
+
+Select your new object (again, you have to click the name itself). This page similarly lets you see some details and configure properties. What we're after is the URL at the bottom of the page, under the label `Link`.
+
+Copy the link and paste it in the URL bar of a new tab in your browser. The canyon should start loading, albeit slowly. Why so slow? Well, we did place it in a region half-way 'round the world. 
+
+Let's try to quantify that speed. Edit the last line in the included shell script `lat-test.sh`, using a command line text editor or IDE, and drop in the copied link. If you haven't already, enter your environment now by running:
+
+```
+ $ pipenv shell
+```
+
+Make sure you have the proper dependencies from the `Pipfile` with the command:
+
+```
+$ pipenv install
+```
+
+You may need to make the script executable. While in the same directory as the script is located, run:
+
+```
+$ chmod +x lat-test.sh
+``` 
+
+Now execute it but running:
+
+```
+./lat-test.sh
+```
+
+- What is the output?
```

Comment:
Thanks for adding this, pretty neat. I looked into curl and it'd be neat to add `speed_download` too. Would be interesting to compare the download speeds themselves between Sydney and Virginia